stock_owner_restriction/models/stock_quant.py

```diff
@@ -25,20 +26,33 @@ def _gather(
             owner_id=owner_id,
             strict=strict,
         )
-        restricted_owner_id = self.env.context.get("force_restricted_owner_id", None)
-        if owner_id is None or restricted_owner_id is None:
+        restricted_owner = self.env.context.get("force_restricted_owner_id", None)
+        if owner_id is None or restricted_owner is None:
             return records
         return records.filtered(
-            lambda q: q.owner_id == (restricted_owner_id or self.env["res.partner"])
+            lambda q: q.owner_id == (restricted_owner or self.env["res.partner"])
         )
 
     @api.model
     def read_group(
         self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True
     ):
-        restricted_owner_id = self.env.context.get("force_restricted_owner_id", None)
-        if restricted_owner_id is not None:
-            domain = expression.AND([domain, [("owner_id", "=", restricted_owner_id)]])
+        owner_in_domain = any(t[0] == "owner_id" for t in domain)
+        if not owner_in_domain:
+            restricted_owner = self.env.context.get("force_restricted_owner_id", None)
+            if restricted_owner is not None:
+                domain = expression.AND(
+                    [
+                        domain,
+                        [
+                            (
+                                "owner_id",
+                                "=",
+                                restricted_owner and restricted_owner.id or False,
+                            )
+                        ],
+                    ]
+                )
```

Comment:
Let's save a few lines.

```suggestion
                owner_domain = [
                    ("owner_id", "=", restricted_owner and restricted_owner.id or False),
                ]
                domain = expression.AND([domain, owner_domain])
```